src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java

```diff
@@ -580,6 +580,11 @@ private void fulfill(@NotNull final Button button)
             {
                 itemStack = inventory.getStackInSlot(InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(inventory), requestPredicate));
             }
+
+            for (int checkInv = 0; checkInv <= 3; checkInv++)
+                if (itemStack.isItemEqual(inventory.armorInventory.get(checkInv))) {
```

Comment:
But like this, I think it won't work if the player has the armor in his armor slots + the armor in his inventory.
+ Can you put brackets around the for and the brackets for the if on an extra line?

Reply:
In my tests it seems to work with armor equipped and in the inventory